src/tools/rest-backend/tests/test_service_configengine.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file
+ *
+ * @brief tests for the configformat model
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include <regex>
+
+#include <gtest/gtest.h>
+
+#include <service.hpp>
+
+/**
+ * TESTS for kdbrest::models::ConfigFormat
+ */
+
+TEST (kdbrestServiceConfigengineTest, ArrayKeyRegexCheck)
+{
+
+	std::regex regex (kdbrest::service::REGEX_CHECK_KEY_IS_ARRAY);
+
+	ASSERT_FALSE (std::regex_match ("firstpart", regex));
+	ASSERT_FALSE (std::regex_match ("firstpart.second", regex));
+	ASSERT_FALSE (std::regex_match ("firstpart.second.third", regex));
+
+	ASSERT_TRUE (std::regex_match ("#0", regex));
+	ASSERT_TRUE (std::regex_match ("#5", regex));
+	ASSERT_TRUE (std::regex_match ("#_12", regex));
+	ASSERT_TRUE (std::regex_match ("#_2038", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.#0", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.#_12", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#0", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#0.with.trailing.path", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12.with.trailing.path", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12.with.trailing.array.#0", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12.with.trailing.array.#_12", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12.with.trailing.array.#0.and.path", regex));
+	ASSERT_TRUE (std::regex_match ("firstpart.second.third.#_12.with.trailing.array.#_12.and.path", regex));
+}
+
+TEST (kdbrestServiceConfigengineTest, TransformKeysetToJsonValue)
```

Comment:
Can you add another example which has spec keys? (and default values, overrides and so on)

By exporting with the c plugin can you can get code that represents the live config you use. It would be good to have such a test case.